src/transformers/generation/logits_process.py

```diff
@@ -1702,3 +1702,35 @@ def __call__(self, input_ids, scores):
         unconditional_logits = torch.nn.functional.log_softmax(logits[:, -1], dim=-1)
         out = self.guidance_scale * (scores - unconditional_logits) + unconditional_logits
         return out
+
+
+class EarlyStoppingLogitsProcessor(LogitsProcessor):
```

Comment:
Two comments on this line:
________________________________
"early_stopping" is already a flag in `generate` ([docs](https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.early_stopping)), so we should pick a different name here to avoid a nomenclature clash 🤗 

Perhaps something like `EOSPrioritizerLogitsProcessor`, or something similar
________________________________

Since the docstring says "EOS token is sampled". If this class is exclusively used with `do_sample=True`, then this should be subclassed from `LogitsWarper`. Otherwise, we should rewrite the docs to "selected", instead of "sampled"

Reply:
Thanks! This class indeed only works when used with `do_sample=True`, so I'll update the subclass to `LogitsWarper`. Also, consolidating the comments on the naming, I can rename this class to `BarkEOSPrioritizerLogitsWarper`. Just wanted to be sure that we're aligned with @ylacombe as well?

Reply:
A QQ here! why would it work only when `do_sample=True` ? This should also happen in greedy decoding right ? since it's a stopping criteria (even though we're using `LogitsProcessor`) ?

Reply:
I don't know what greedy decoding means so I'll need to read up on that. In the meantime, I've run `BarkModelIntegrationTests.test_generate_end_to_end_with_sub_models_args` without the `do_sample=True` flag and they still pass. I could switch back to `LogitsProcessor` and change the docs wording to `"selected"` instead of `"sampled"`.

Reply:
> I could switch back to LogitsProcessor and change the docs wording to "selected" instead of "sampled"

That seems the wisest solution for me :)

Reply:
Done. Thanks again for your guidance 🙏